examples/custom-client-cxx/main.cc

```diff
@@ -67,7 +67,26 @@ int main(int argc, char **argv) {
       }
 
       auto latest = res.GetLatest();
-      LOG_INFO << "Found Latest Target: " << latest.Name();
+      if (client.IsRollback(latest)) {
+        LOG_INFO << "Latest Target is marked for causing a rollback and won't be installed: " << latest.Name();
+      } else {
+        LOG_INFO << "Found Latest Target: " << latest.Name();
+      }
+
+      if (client.IsRollback(latest) && current.Name() == latest.Name()) {
+        // Handle the case when Apps failed to start on boot just after an update.
+        // This is only possible with `pacman.create_containers_before_reboot = 0`.
+        LOG_INFO << "The currently booted Target is a failing Target, finding Target to rollback to...";
+        const TufTarget rollback_target = client.GetRollbackTarget();
+        if (rollback_target.IsUnknown()) {
+          LOG_ERROR << "Failed to find Target to rollback to after failure to start Apps at boot of a new sysroot";
+          std::this_thread::sleep_for(std::chrono::seconds(interval));
+          continue;
+        }
+        latest = rollback_target;
+        LOG_INFO << "Rollback Target is " << latest.Name();
```

Comment:
Maybe write "Rolling back or will roll back..." to inform about the further action the client is going to do.

Reply:
i was on the fence on this one as well. The very next log message is almost always "Updating from X to Y". So I thought when you read them in sequence it kind of made sense.